src/overlay.rs

```diff
@@ -60,23 +79,25 @@ where
             tracing::trace!(?key, value = ?hex::encode(&value), "read from cache");
             Ok(Some(value.clone()))
         } else {
-            match self.tree().get(key, self.version).await {
-                Ok(Some(value)) => {
-                    tracing::trace!(version = ?self.version, ?key, value = ?hex::encode(&value), "read from tree");
-                    Ok(Some(value))
-                }
-                Ok(None) => {
-                    tracing::trace!(version = ?self.version, ?key, "key not found in tree");
-                    Ok(None)
-                }
-                // This allows for using the Overlay on an empty database without errors
-                Err(e) if e.downcast_ref::<MissingRootError>().is_some() => {
-                    tracing::trace!(version = ?self.version, "no data available at this version");
-                    Ok(None)
-                }
-                Err(e) => Err(e),
-            }
+            self.get_from_tree(key).await
+        }
+    }
+
+    /// Gets a value by key alongside an ICS23 existence proof of that value.
+    ///
+    /// This method does *not* reflect results of any pending writes to the WriteOverlay.
+    #[instrument(name = "WriteOverlay::get_with_proof", skip(self, key))]
+    pub async fn get_with_proof(
+        &mut self,
+        key: Vec<u8>,
```

Comment:
NOTE: this interface is for the *raw* key, not the KeyHash. This is because ICS23 existence proofs need the raw key, not just the key hash.